Add VerticalSizer for track header

pierricgimmig opened the pull request:
Add vertical sizer class to resize new track header. The cursor
changes to horizontal arrows when the mouse is in the sizer.
We keep a minimum width equal to the height of the timeline
for symmetry, to keep the triangle toggles visible, and to keep 
the track moving feature available. 

Also, set right margin to 0, but keep logic just in case, and tweak 
distance between timeline and tracks for more stream lined look.

![image](https://user-images.githubusercontent.com/3687222/213963150-d41ab2ee-4474-48fe-a17e-43871397a1be.png)

![image](https://user-images.githubusercontent.com/3687222/213963249-64dd721f-31ea-44e4-bbda-2cf99eaf8b5d.png)

reichlfl commented on src/OrbitGl/VerticalSizer.cpp:
```diff
@@ -0,0 +1,56 @@
+
+void VerticalSizer::OnDrag(int x, int y) {
+  on_drag_callback_(x, y);
+  RequestUpdate(RequestUpdateScope::kDraw);
```
I would argue that in this case, `RequestUpdate` should not happen as well. It's the responsibility of the code in `on_drag_callback_` to decide wether an update is needed or not as the sizer is not updating it's own position.

pierricgimmig replied:
Done

reichlfl commented on src/OrbitGl/TimeGraph.cpp:
```diff
@@ -887,6 +895,13 @@ void TimeGraph::DoDraw(orbit_gl::PrimitiveAssembler& primitive_assembler,
   }
+
+  // Track header sizer.
+  float sizer_width = layout_->GetSpaceBetweenTracks();
+  float header_width = layout_->GetTrackHeaderWidth();
+  track_header_sizer_->SetWidth(sizer_width);
+  track_header_sizer_->SetHeight(GetHeight());
+  track_header_sizer_->SetPos(GetPos()[0] + header_width - sizer_width, GetPos()[1]);
```
`DoDraw` and `DoUpdatePrimitives` must not call any methods that result in a redraw. Move this to `TimeGraph::DoUpdateLayout`.

pierricgimmig replied:
Done

reichlfl reviewed (changes requested):
This is really cool, thanks! I left a couple of comments regarding update / redraw conventions that need to be followed, otherwise we risk some changes not showing up until the next frame.

pierricgimmig commented on src/OrbitGl/VerticalSizer.cpp:
```diff
@@ -0,0 +1,56 @@
+void VerticalSizer::DoDraw(PrimitiveAssembler& primitive_assembler, TextRenderer& text_renderer,
+                           const DrawContext& draw_context) {
+  CaptureViewElement::DoDraw(primitive_assembler, text_renderer, draw_context);
+  Quad right_margin_box = MakeBox(GetPos(), GetSize());
```
copy-paste is not my friend, changed to `box`.

pierricgimmig commented on src/OrbitGl/include/OrbitGl/VerticalSizer.h:
```diff
@@ -0,0 +1,46 @@
+
+namespace orbit_gl {
+
+class VerticalSizer : public orbit_gl::CaptureViewElement,
```
Done.

pierricgimmig commented on src/OrbitGl/VerticalSizer.cpp:
```diff
@@ -0,0 +1,56 @@
+  if (QGuiApplication::instance() != nullptr) {
+    QGuiApplication::restoreOverrideCursor();
+  }
+  RequestUpdate(RequestUpdateScope::kDraw);
```
Removed

pierricgimmig commented on src/OrbitGl/VerticalSizer.cpp:
```diff
@@ -0,0 +1,56 @@
+  if (QGuiApplication::instance() != nullptr) {
+    QGuiApplication::setOverrideCursor(Qt::SizeHorCursor);
+  }
+  RequestUpdate(RequestUpdateScope::kDraw);
```
Removed

pierricgimmig commented on src/OrbitGl/include/OrbitGl/VerticalSizer.h:
```diff
@@ -0,0 +1,46 @@
+  void OnDrag(int /*x*/, int /*y*/) override;
+
+  [[nodiscard]] float GetHeight() const override { return height_; }
+  void SetHeight(float height) { height_ = height; }
```
Done. Btw, there is an asymmetry with the SetWidth method which does this boilerplate, we should probably fix SetHeight so it behaves similarly.

pierricgimmig commented:
> This is really cool, thanks! I left a couple of comments regarding update / redraw conventions that need to be followed, otherwise we risk some changes not showing up until the next frame.

Thanks for the review! I addressed all the comments, please take another look. Cheers!

florian-kuebler commented on src/OrbitGl/include/OrbitGl/TimeGraph.h:
```diff
@@ -200,6 +204,7 @@ class TimeGraph : public orbit_gl::CaptureViewElement, public orbit_gl::Timeline
 
   std::shared_ptr<orbit_gl::GlSlider> horizontal_slider_;
   std::shared_ptr<orbit_gl::GlSlider> vertical_slider_;
+  std::shared_ptr<orbit_gl::VerticalSizer> track_header_sizer_;
```
Why does it need to be a shared pointer?

pierricgimmig replied:
I think because the picking system expects it.

florian-kuebler replied:
Ah I see. There it probably also is added. Missed that. Thanks

reichlfl replied:
Yeah, this is needed due to picking, we may want to clean that up...

florian-kuebler reviewed (approved):
LGTM. Only one question.

reichlfl reviewed (approved):
Thanks for the changes, LGTM!
